### ISSUE TITLE
Build fails

### ISSUE THREAD

**Issue by Peeja** [tgdwyer/WebCola#232]
On v3.3.5 (de8ebf0):

```
WebCola$ grunt
Running "copy:qunit" (copy) task
Copied 2 files

Running "ts:commonjs" (ts) task
Compiling...
Using tsc v2.4.1
>> Warning: created WebCola/.baseDir.ts file because `outDir` was specified in the tsconfig.json, but not `rootDir`.  Add `rootDir`  in the `compilerOptions` section to fix this warning.
WebCola/src/adaptor.ts(3,18): error TS2415: Class 'LayoutAdaptor' incorrectly extends base class 'Layout'.
  Types of property 'on' are incompatible.
    Type '(eventType: string | EventType, listener: () => void) => this' is not assignable to type '(e: string | EventType, listener: (event: Event) => void) => this'.
      Types of parameters 'listener' and 'listener' are incompatible.
WebCola/src/d3adaptor.ts(35,9): error TS2322: Type 'D3StyleLayoutAdaptor' is not assignable to type 'Layout & ID3StyleLayoutAdaptor'.
  Type 'D3StyleLayoutAdaptor' is not assignable to type 'Layout'.
    Types of property 'on' are incompatible.
      Type '(eventType: string | EventType, listener: () => void) => D3StyleLayoutAdaptor' is not assignable to type '(e: string | EventType, listener: (event: Event) => void) => Layout'.
        Types of parameters 'listener' and 'listener' are incompatible.
WebCola/src/d3adaptor.ts(37,5): error TS2322: Type 'D3StyleLayoutAdaptor' is not assignable to type 'Layout & ID3StyleLayoutAdaptor'.
  Type 'D3StyleLayoutAdaptor' is not assignable to type 'Layout'.
    Types of property 'on' are incompatible.
      Type '(eventType: string | EventType, listener: () => void) => D3StyleLayoutAdaptor' is not assignable to type '(e: string | EventType, listener: (event: Event) => void) => Layout'.
        Types of parameters 'listener' and 'listener' are incompatible.
WebCola/src/d3v3adaptor.ts(10,18): error TS2415: Class 'D3StyleLayoutAdaptor' incorrectly extends base class 'Layout'.
  Types of property 'on' are incompatible.
    Type '(eventType: string | EventType, listener: () => void) => this' is not assignable to type '(e: string | EventType, listener: (event: Event) => void) => this'.
      Types of parameters 'listener' and 'listener' are incompatible.
WebCola/src/d3v4adaptor.ts(14,14): error TS2415: Class 'D3StyleLayoutAdaptor' incorrectly extends base class 'Layout'.
  Types of property 'on' are incompatible.
    Type '(eventType: string | EventType, listener: () => void) => this' is not assignable to type '(e: string | EventType, listener: (event: Event) => void) => this'.
      Types of parameters 'listener' and 'listener' are incompatible.

>> 5 non-emit-preventing type warnings  
>> Error: tsc return code: 2
Warning: Task "ts:commonjs" failed. Use --force to continue.

Aborted due to warnings.
```

**Comment by tamird** [comment]
Looks like this is due to new strictness in typescript 2.4.

**Comment by tgdwyer** [comment]
fixed v3.3.6 47e5fd82986861bc147b77f7f066cf50611d8fa2
